pkg/cloudclient/gcp/private.go

```diff
@@ -201,3 +283,245 @@ func getIPAddressesFromService(svc *corev1.Service) ([]string, error) {
 
 	return ips, nil
 }
+
+func (c *Client) removeLoadBalancerFromMasterNodes(ctx context.Context, kclient client.Client) (string, error) {
+	region, err := getClusterRegion(kclient)
+	if err != nil {
+		return "", err
+	}
+
+	listCall := c.computeService.ForwardingRules.List(c.projectID, region)
+	response, err := listCall.Do()
+	if err != nil {
+		return "", err
+	}
+
+	masterList, err := baseutils.GetMasterMachines(kclient)
+	if err != nil {
+		return "", err
+	}
+	var intIPAddress, lbName string
+	for _, lb := range response.Items {
+		// This list of forwardingrules (LBs) includes any service LBs
+		// for application routers so check the port range to identify
+		// the external API LB.
+		if lb.LoadBalancingScheme == "EXTERNAL" && lb.PortRange == "6443-6443" {
+			//delete the LB and remove it from the masters
+			lbName = lb.Name
+			_, err := c.computeService.ForwardingRules.Delete(c.projectID, region, lbName).Do()
+			if err != nil {
+				return "", fmt.Errorf("Failed to delete ForwardingRule for external load balancer %v: %v", lb.Name, err)
+			}
+			err = removeGCPLBFromMasterMachines(kclient, lbName, masterList)
+			if err != nil {
+				return "", err
+			}
+		}
+		// we need this to update DNS
+		if lb.LoadBalancingScheme == "INTERNAL" && lb.BackendService != "" {
+			// Unlike AWS, GCP NLBs don't have automatically assigned A records, just an external IP address
+			// Save the internal NLB's IP Address in order to update the API's A record in the public DNS zone.
+			intIPAddress = lb.IPAddress
+		}
+	}
+	return intIPAddress, nil
+}
+
+func getClusterRegion(kclient client.Client) (string, error) {
+	infra, err := baseutils.GetInfrastructureObject(kclient)
+	if err != nil {
+		return "", err
+	}
+	return infra.Status.PlatformStatus.GCP.Region, nil
```

Comment:
This can be addressed later, but the base `utils` package also has a `GetClusterRegion` function.  It currently assumes AWS, so it needs extended for GCP with a "switch" statement on the platform type, but would be preferable to use that instead.